packages/docgen/src/engine.js

```diff
@@ -0,0 +1,56 @@
+/**
```

Comment:
I wonder if we should have avoided placing these files in a directory called `src/`, because for every other package we use the `src` directory to indicate pre-transpiled code, with the expectation there'd be a complementing `build` directory as output. I encountered this today when auditing our transpiled packages for `@babel/runtime` and became confused when this package was not respecting the convention.

In other words, I see no reason why all of these files couldn't just live in the root of the package, or maybe we could create a different convention for organizing files within packages which don't need transpiling.

Reply:
In fact, `src/` is used as a trigger to process files via Babel.

You can see that `docgen` includes `build` and `build-module` outputs, even though they're effectively dead code:

https://unpkg.com/@wordpress/docgen@1.0.0/

Related code:

https://github.com/WordPress/gutenberg/blob/9c3dbde9f7585ed776fa52a3720ad581d556f865/bin/packages/build.js#L193-L200

Reply:
We can change this logic and decide whether a given package should be transpiled based on the existance of `module` entry in `package.json` file.

Reply:
I see what you mean. `docgen` never uses transpiled code but it's still published to npm.

Reply:
> In fact, src/ is used as a trigger to process files via Babel.

Didn't know that until now.

> We can change this logic and decide whether a given package should be transpiled based on the existance of module entry in package.json file.

I like this. It allows us to store code within `src/` for complex packages like `docgen` that require multiple files. It's also in line with the idea we discussed somewhere about favoring `package.json` config over maintaining ignore lists.